Makefile

```diff
@@ -201,7 +201,11 @@ UI_ROOT        := $(PKG_ROOT)/ui
 SQLPARSER_ROOT := $(PKG_ROOT)/sql/parser
 
 # Ensure we have an unambiguous GOPATH.
-GOPATH := $(shell $(GO) env GOPATH)
+GOPATH := $(shell readlink -f $(shell $(GO) env GOPATH))
+# GOPATH need not be explicitly set if it is the default $HOME/go
+ifeq ($(strip $(GOPATH)),)
+GOPATH := $(shell readlink -f $$HOME/go)
+endif
```

Comment:
Hmm, I can't reproduce the need for this. `go env GOPATH` does the right thing for me when GOPATH is not set in the environment—and in fact that's why we don't use `GOPATH` directly. Does the following return a different result for you?

```
$ unset GOPATH && go env GOPATH
/home/benesch/go
```

Reply:
```fish
mqudsi@mysql ~/g/s/g/c/cockroach> go version
go version go1.10.1 freebsd/amd64
mqudsi@mysql ~/g/s/g/c/cockroach> set -e GOPATH; go env GOPATH

mqudsi@mysql ~/g/s/g/c/cockroach> env GOPATH="" go env GOPATH
```

Reply:
However, on another machine:

```fish
mqudsi@freebsd ~/g/s/g/c/cockroach> go version
go version go1.10.1 freebsd/amd64
mqudsi@freebsd ~/g/s/g/c/cockroach> env GOPATH="" go env GOPATH
/home/mqudsi/go
```

Not sure what to make of this. A golang bug, perhaps?

Reply:
Check this out:

```
$ HOME= GOPATH= go env GOPATH

$
```

Seems like maybe `$HOME` is unset? Or `os/user.Current` is otherwise bricked on that machine?

Reply:
Nope, `$HOME` is correct. This was a new deployment of both go and cockroach, just installed. cockroach is the first go installation on this machine. Perhaps the clue lies therein somewhere?

Reply:
Ok, I've come up with another way of causing this situation:

```
$ GOROOT=$HOME/go GOPATH= go env GOPATH

$
````

Does that perhaps describe your situation? If not, could you post the full output of `go env`?

Reply:
You nailed it. GOPATH and GOROOT can't both be the same, so that breaks the auto-generation of GOPATH.

I guess the best thing here is to error out if `go env GOPATH` returns an empty result.